File: Portal/src/Datahub.Infrastructure/Queues/Messages/BugReportMessage.cs
using MediatR;

namespace Datahub.Infrastructure.Queues.Messages
{
    /// <summary>
    /// Represents the different types of bug reports. To create a new type,
    /// add a User Story in ADO and grab the reference ID.
    /// </summary>
    public enum BugReportTypes
    {
        SupportRequest = 1230,
        SystemError = 1700,
        InfrastructureError = 1699,
        PythonWorkspaceSyncError = 7023,
    }
    public record BugReportMessage(
        string? UserName,
        string? UserEmail,
        string? UserOrganization,
        string? PortalLanguage,
        string? PreferredLanguage,
        string? Timezone,
        string? Workspaces,
        string? Topics,
        string? URL,
        string? UserAgent,
        string? Resolution,
        string? LocalStorage,
        BugReportTypes BugReportType,
        string Description
    ) : IRequest;
}

File: ResourceProvisioner/src/ResourceProvisioner_PyFunctions/bug_report_message.py
import json
import datetime

class BugReportMessage:
    def __init__(self, UserName, UserEmail, UserOrganization, PortalLanguage, PreferredLanguage, Timezone, Workspaces, Topics, URL, UserAgent, Resolution, LocalStorage, BugReportType, Description):
        self.UserName = UserName
        self.UserEmail = UserEmail
        self.UserOrganization = UserOrganization
        self.PortalLanguage = PortalLanguage
        self.PreferredLanguage = PreferredLanguage
        self.Timezone = Timezone
        self.Workspaces = Workspaces
        self.Topics = Topics
        self.URL = URL
        self.UserAgent = UserAgent
        self.Resolution = Resolution
        self.LocalStorage = LocalStorage
        self.BugReportType = BugReportType
        self.Description = Description

    def to_json(self):
        def json_default(value):
            if isinstance(value, datetime.date):
                return dict(year=value.year, month=value.month, day=value.day)
            else:
                return value.__dict__        
        return json.dumps(self, default=json_default, sort_keys=True, indent=4)